pkg/api/customization/oci/handler.go

```diff
@@ -0,0 +1,238 @@
+package oci
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/rancher/norman/api/access"
+	"github.com/rancher/norman/types"
+	client "github.com/rancher/types/client/management/v3"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/gorilla/mux"
+	"github.com/oracle/oci-go-sdk/common"
+	"github.com/rancher/norman/httperror"
+	"github.com/rancher/rancher/pkg/auth/util"
+	"github.com/rancher/rancher/pkg/ref"
+	v1 "github.com/rancher/types/apis/core/v1"
+	"github.com/rancher/types/apis/management.cattle.io/v3/schema"
+	"github.com/rancher/types/config"
+	"github.com/rancher/types/namespace"
+
+	"encoding/json"
+)
+
+type Credentials struct {
+	Tenancy              string `json:"tenancyOCID"`
+	User                 string `json:"userOCID"`
+	Region               string `json:"region"`
+	FingerPrint          string `json:"fingerprint"`
+	PrivateKey           string `json:"privateKey"`
+	PrivateKeyPassphrase string `json:"privateKeyPassphrase"`
+	Compartment          string `json:"compartmentOCID"`
+}
+
+// Cloud Credential Secret Fields
+var requiredDataFields = map[string]string{
+	"fingerprint":        "ocicredentialConfig-fingerprint",
+	"tenancyId":          "ocicredentialConfig-tenancyId",
+	"userId":             "ocicredentialConfig-userId",
+	"privateKeyContents": "ocicredentialConfig-privateKeyContents",
+}
+
+type Handler struct {
+	Action        string
+	schemas       *types.Schemas
+	secretsLister v1.SecretLister
+}
+
+func NewOCIHandler(scaledContext *config.ScaledContext) http.Handler {
+	return &Handler{
+		schemas:       scaledContext.Schemas,
+		secretsLister: scaledContext.Core.Secrets(namespace.GlobalNamespace).Controller().Lister(),
+	}
+}
+
+func (handler *Handler) ServeHTTP(writer http.ResponseWriter, req *http.Request) {
+
+	writer.Header().Set("Content-Type", "application/json")
+
+	// New credential every call
+	creds := Credentials{}
+	errCode, err := handler.extractCreds(writer, req, &creds)
+	if err != nil {
+		util.ReturnHTTPError(writer, req, errCode.Status, err.Error())
+		return
+	}
+
+	err = handler.validateCreds(&creds)
+	if err != nil {
+		util.ReturnHTTPError(writer, req, httperror.ServerError.Status, err.Error())
```

Comment:
This needs to be updated to `InvalidBodyContent`, a user passing in bad/incomplete data is not a server error.